modules/system/classes/CompilableAssets.php

```diff
@@ -0,0 +1,340 @@
+<?php namespace System\Classes;
+
+use Cms\Classes\Theme;
+use System\Classes\PluginManager;
+use Winter\Storm\Exception\SystemException;
+use Winter\Storm\Filesystem\PathResolver;
+use Winter\Storm\Support\Facades\Config;
+use Winter\Storm\Support\Facades\File;
+use Winter\Storm\Support\Str;
+
+/**
+ * Compilable assets using for Node.js compilation and processing.
+ *
+ * This works similar to the `System\Classes\CombineAssets` class in that it allows modules, plugins and themes to
+ * register configurations that will be passed on to Node.js based compilers for processing.
+ *
+ * @package winter\wn-system-module
+ * @author Ben Thomson <git@alfreido.com>, Jack Wilkinson <me@jackwilky.com>
+ * @author Winter CMS
+ */
+class CompilableAssets
+{
+    use \Winter\Storm\Support\Traits\Singleton;
+
+    /**
+     * The filename that stores the package definition.
+     */
+    protected string $packageJson = 'package.json';
+
+    /**
+     * List of package types and registration methods
```

Comment:
Sure is painful, but let's start describing array shapes in the event I decide to tackle PHPStan with the modules.
```suggestion
     * @var array<string, array<string, string>> List of package types and registration methods
```